### DISCUSSION TITLE
Deep Mobile Integration: Enhancing Bevy's Interoperability with iOS and Android

### DISCUSSION THREAD

**Discussion by akfpnd** [bevyengine/bevy#10900] [Ideas]
Hello Bevy Community,

I'm reaching out to discuss and seek guidance on a crucial aspect of game development using Bevy: deep mobile integration. As we're all aware, Bevy shows incredible potential in game development, but I've noticed a gap in its integration capabilities with mobile platforms, specifically when it comes to interoperability with Kotlin in Android Studio and some gaps in Xcode and Swift.

While exploring online resources, I realized there's a lack of comprehensive examples or documentation on how to seamlessly integrate Bevy as a library in mobile projects. This integration is essential for leveraging platform-specific features like accelerometer, gyroscope support, in-app purchases, and integration with services like Google Play Games and Apple's equivalent.

Notably, engines like Unity and Unreal provide the facility to export projects directly to Android Studio and Xcode, aligning well with the mobile ecosystem. Although Bevy offers some support for Xcode, there's a noticeable absence of equivalent support for Android Studio. This limitation hinders developers who wish to create more immersive and integrated mobile experiences using Bevy.

A detailed guideline on setting up Bevy as a library in Android Studio through JNI (Java Native Interface) would be immensely helpful. Additionally, more robust examples and documentation for iOS integration, especially detailing the interop with Swift, are necessary. The current examples and resources don't fully cover these aspects.

In my opinion, addressing these issues is a crucial step towards professional mobile game development with Bevy. It would not only enhance the engine's capabilities but also broaden its appeal to a larger segment of developers who are interested in mobile game development.

I'm eager to hear from the community and the developers behind Bevy. Any insights, guidance, or plans regarding this integration would be greatly appreciated. Let's work together to make Bevy a more versatile and mobile-friendly game development engine.

Thank you for your time and consideration.

**Comment by alice-i-cecile** [comment]
Definitely agree: mobile support is an area where I've both noticed Bevy lacking and one where I think there's a huge market demand for better game engines.

Better documentation (targeted at a new-to-mobile developer to start) is a huge part of both improving the Bevy user experience and making it easier to contribute there.

**Reply by akfpnd** [reply]
@alice-i-cecile could you help how to get started building Bevy as a library? It was mentioned in documentation but didn't elaborate how exactly setup. If Bevy using Winit it utilising `GameActivity`/`NativeActivity` under the hood is it possible to expose this Activity in android project from builded library to extend with Activity in Kotlin to establish minimum interop  through FFI/JNI? 
Or you have better idea how integration with mobile may look to provide access to all mobile specific features on each platform?

**Reply by alice-i-cecile** [reply]
That's part of the problem: I don't know and there's no docs! I'm very familiar with Bevy, but not at all with mobile development currently.

**Comment by atonamy** [comment]
@alice-i-cecile I found workaround [here](https://github.com/jinleili/bevy-in-app)
Huge credit to @jinleili 🙏
I hope in the future Bevy support such feature officially and provide example with documentation and less boilerplate code. 

cc @ickshonpe
